packages/gatsby/src/redux/persist.ts

```diff
@@ -67,10 +70,29 @@ export function readFromCache(): ICachedReduxState {
 
   obj.nodes = new Map(nodes)
 
+  // Note: at 1M pages, this will be 1M/chunkSize chunks (ie. 1m/10k=100)
+  const pagesChunks = globSync(
+    reduxChunkedPagesFilePrefix(reduxCacheFolder) + `*`
+  ).map(file => v8.deserialize(readFileSync(file)))
+
+  const pages: Array<[string, IGatsbyPage]> = [].concat(...pagesChunks)
+
+  if (!pagesChunks.length) {
```

Comment:
Potentially problematic - this will discard state if there are 0 pages - do we want to optimize for those cases? (I don't even know if gatsby actually "works" if there are 0 pages TBH)

Reply:
When way to go about this is to add `pagesCount` to `redux.rest` part and compare it I guess? Cache will get invalidated anyway after upgrade due to `pagesChunk.length` being 0 so I think cache shape "version" could be handled gracefully

Reply:
If there were no pages then it's probably no big deal to bust the cache anyways.

Reply:
As per off-thread discussion. I think you can merge this PR in a patch without this check, if you wanted to and then add the check in a minor bump later.